coord/src/rmr/redise.c

```diff
@@ -5,38 +5,191 @@
  */
 
 #include "redise.h"
-#include "redise_parser/parse.h"
+
+typedef struct {
+  int startSlot;
+  int endSlot;
+  MRClusterNode node;
+} RLShard;
+
+static void MRTopology_AddRLShard(MRClusterTopology *t, RLShard *sh) {
+
+  int found = -1;
+  for (int i = 0; i < t->numShards; i++) {
+    if (sh->startSlot == t->shards[i].startSlot && sh->endSlot == t->shards[i].endSlot) {
+      found = i;
+      break;
+    }
+  }
+
+  if (found >= 0) {
+    MRClusterShard_AddNode(&t->shards[found], &sh->node);
```

Comment:
If the shard already exists, it necessarily mean that we are adding a replica? If so, can this be validated?

Reply:
You mean to fail if it’s a master? I think we can add the replica before the master. We can validate that for each shard there is one master.
We can also discuss more validation like that there is only one self, that all the ids are unique, that one of the shards is self